src/Compilers/CSharp/Portable/Symbols/OverriddenOrHiddenMembersHelpers.cs

```diff
@@ -921,7 +921,8 @@ internal static bool RequiresExplicitOverride(this MethodSymbol method, out bool
                 return false;
 
             // See https://github.com/dotnet/roslyn/issues/45453. No need to warn when the runtime
-            // supports covariant returns because the methodimpl is unambiguously understood by the runtime.
+            // supports covariant returns because any methodimpl we produce to identify the specific
+            // overridden method is unambiguously understood by the runtime.
             if (method.ContainingAssembly.RuntimeSupportsCovariantReturnsOfClasses)
                 return true;
```

Comment:
>return true; [](start = 16, length = 12)

I do not think we should get here for scenarios in the first several tests. Whether runtime supports covariant returns makes no difference for those cases, they cannot be helped by adding explicit MethodIml in metadata.

Reply:
This code is not changed in this PR. In any case, reaching here is not harmful and reflect the design intent.

---
In reply to: [452978462](https://github.com/dotnet/roslyn/pull/45849#discussion_r452978462) [](ancestors = 452978462)

Reply:
> This code is not changed in this PR.

I do not think this is relevant. This code was added in the previous PR and this PR is a follow up on an issue that was identified during the previous PR.

---
In reply to: [453764727](https://github.com/dotnet/roslyn/pull/45849#discussion_r453764727) [](ancestors = 453764727,452978462)

Reply:
This PR addresses the issue identified in the previous PR, which is eliminating cascaded diagnostics and additional testing.

---
In reply to: [453809152](https://github.com/dotnet/roslyn/pull/45849#discussion_r453809152) [](ancestors = 453809152,453764727,452978462)